ksqldb-functional-tests/src/test/resources/query-validation-tests/none.json

```diff
@@ -0,0 +1,181 @@
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "The 'NONE' format can only be used when no columns are defined. Got: [`K` STRING KEY]"
+      }
+    },
+    {
+      "name": "as value format with value columns",
+      "statements": [
+        "CREATE STREAM INPUT (foo INT) WITH (kafka_topic='input_topic', value_format='NONE');"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "The 'NONE' format can only be used when no columns are defined. Got: [`FOO` INTEGER]"
+      }
+    },
+    {
+      "name": "in CSAS from keyless stream",
+      "statements": [
+        "CREATE STREAM INPUT (foo INT) WITH (kafka_topic='input_topic', key_format='KAFKA', value_format='JSON');",
+        "CREATE STREAM OUTPUT WITH(key_format='NONE') AS SELECT * FROM INPUT;"
+      ],
+      "inputs": [
+        {"topic": "input_topic", "key": "ignored", "value": {"FOO": 10}},
+        {"topic": "input_topic", "key": null, "value": null}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": null, "value": {"FOO": 10}},
+        {"topic": "OUTPUT", "key": null, "value": null}
+      ]
+    },
+    {
+      "name": "in CSAS partitioning by null",
+      "statements": [
+        "CREATE STREAM INPUT (ID INT KEY, foo INT) WITH (kafka_topic='input_topic', key_format='KAFKA', value_format='JSON');",
+        "CREATE STREAM OUTPUT WITH(key_format='NONE') AS SELECT * FROM INPUT PARTITION BY null;"
+      ],
+      "inputs": [
+        {"topic": "input_topic", "key": 9, "value": {"FOO": 10}},
+        {"topic": "input_topic", "key": null, "value": null}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": null, "value": {"FOO": 10, "ID": 9}},
+        {"topic": "OUTPUT", "key": null, "value": null}
+      ]
+    },
+    {
+      "name": "explicitly set in CSAS with key",
+      "statements": [
+        "CREATE STREAM INPUT (foo INT, bar INT) WITH (kafka_topic='input_topic', key_format='KAFKA', value_format='JSON');",
+        "CREATE STREAM OUTPUT WITH (key_format='NONE') AS SELECT * FROM INPUT PARTITION BY BAR;"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "The 'NONE' format can only be used when no columns are defined. Got: [`BAR` INTEGER KEY]"
+      }
+    },
+    {
+      "name": "explicitly set in CTAS",
+      "statements": [
+        "CREATE STREAM INPUT (foo INT, bar INT) WITH (kafka_topic='input_topic', key_format='KAFKA', value_format='JSON');",
+        "CREATE TABLE OUTPUT WITH (key_format='NONE') AS SELECT BAR AS K, COUNT() FROM INPUT GROUP BY BAR;"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "The 'NONE' format can only be used when no columns are defined. Got: [`K` INTEGER KEY]"
```

Comment:
Not directly related to this PR: I notice the printing of the schema doesn't distinguish between `KEY` and `PRIMARY KEY`. What was the reasoning for that? Is it just easier to not worry about passing in information about source type to the schema, or is the current behavior preferred outside of the logistical concern too?

Reply:
The text is coming from `LogicalSchema` / `Column`, which do not differentiate between primary key and key. 

We could just strip off the KEY all together, but I don't really see it as a big thing.  The error message has enough information that the user can correct the error.